### ISSUE TITLE
Mac/BSD, for sys.stdout: "When appending to an avro file you must use the 'a+' mode"

### ISSUE THREAD

**Issue by jquast** [fastavro/fastavro#330]
How curious how we learn how these platforms behave differently,

mac
---

```
$ python -V
Python 3.6.5
$ python -c 'import sys;print(sys.stdout.buffer.tell())'
1151335
```

linux
-----

```
$ python -V
Python 3.6.6
$ python -c 'import sys;print(sys.stdout.buffer.tell())'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
OSError: [Errno 29] Illegal seek
```

This platform-specific behavior impedes an edge-case feature: what if we wanted to use fastavro to stream directly to `sys.stdout`? Such as in a classic unix pipe -- this works perfectly ok on Linux, great! 

Example:

```
In [3]: import sys, fastavro                                                                                         

In [4]: fout = fastavro.write.Writer(sys.stdout.buffer, schema=fastavro.parse_schema({'name': 'ohhai', 'type': 'record', 'fields': [ {'name': 'temp', 'type': 'int'}]})); fout.flush()                                            
Objavro.codenullavro.schema�{"type": "record", "name": "ohhai", "fields": [{"name": "temp", "type": "int"}], "__fastavro_parsed": true}D�A�7G+*+w�9ɨ�
```

Cool :) But on Mac, this isn't possible, the code introduced in https://github.com/fastavro/fastavro/commit/aaf02423#diff-fbbdb913f5928803608b2925d15d137fR49 raises the ValueError,

related comment, https://github.com/fastavro/fastavro/issues/276#issuecomment-424309799

```
In [2]: fout = fastavro.write.Writer(sys.stdout.buffer, schema=fastavro.parse_schema({'name': 'ohhai', 'type': 'record'
   ...: , 'fields': [ {'name': 'temp', 'type': 'int'}]})); fout.flush()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-dc0833ee58e4> in <module>
----> 1 fout = fastavro.write.Writer(sys.stdout.buffer, schema=fastavro.parse_schema({'name': 'ohhai', 'type': 'record', 'fields': [ {'name': 'temp', 'type': 'int'}]})); fout.flush()

~/Code/fastavro/fastavro/_write.pyx in fastavro._write.Writer.__init__()

~/Code/fastavro/fastavro/_six.pyx in fastavro._six.py3_appendable()

ValueError: When appending to an avro file you must use the 'a+' mode, not just 'a'
```

**Comment by jquast** [comment]
For this error I am happy to suggest a fix, but I'm not sure how, the current technique is creative, I think the solution would be even more creative, possibly rejected :) It's also really hard to write a test case as it is a platform-specific behavior, anyway I'm happy to suggest something with some mock tests, I just need a little direction please!

**Comment by scottbelden** [comment]
Hmmm. This is quite frustrating. I think perhaps the best way to handle this (without adding a ton of platform specific logic) would be to say that it doesn't make sense to append to `sys.stdout`. If you think that's a safe assumption, then we could add a check like:

```python
if file_like == sys.stdout.buffer:
    return False
```

to the beginning of the `appendable` functions in the `six` modules we have.

Do you have any thoughts about this?

**Comment by jquast** [comment]
That seems perfectly fine, I was thinking something along the same lines, I'll  come up with a patch to include integration tests, I see that Travis-CI has the osx agents so it should verify just fine, thanks !!

**Comment by jquast** [comment]
I suggested fix #342 please enjoy!
